### PR TITLE
RISC-V: assorted changes for OpenJ9 JIT

### PR DESCRIPTION
**Pull request by janvrany**
This PR adds a few changes here and there required by OpenJ9 JIT component

### CODE REVIEW

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Just a comment for this PR and any future ones you may be making as you go about implementing an OpenJ9 JIT for RISC-V.  If a commit is truly language-agnostic (ideally they should all be :-) ) and benefits all consumers of OMR then you don't need to call out OpenJ9 in your commit message unless it is particularly relevant.

For example, in the commit message for c4b591253b642135ba3ce22307aeb52120e323b6  you don't need to reference OpenJ9 as an example.

File: compiler/riscv/codegen/OMRLinkage.cpp [hunk 1]
```diff
@@ -161,4 +161,4 @@ OMR::RV::Linkage::numArgumentRegisters(TR_RegisterKinds kind)
       default:
          return 0;
       }
-   }
\ No newline at end of file
+   }
```
[the remark this answers is not in the document]
Comment (janvrany) [reply compiler/riscv/codegen/OMRLinkage.cpp:164]: Fixed in  90b02e8

File: compiler/env/Processors.hpp [hunk 2]
```diff
@@ -159,7 +164,12 @@ enum TR_Processor
    TR_LastX86Processor,
    TR_X86ProcessorAMDOpteron = TR_LastX86Processor,
 
-   TR_LastProcessor = TR_LastX86Processor,
+   TR_FirstRISCVProcessor,
+   TR_RV645G,
```
[the remark this answers is not in the document]
Comment (janvrany) [reply compiler/env/Processors.hpp]: Good catch! It is not, it's a typo followed by copy-paste. Fixed in a24d33e. 
Actually, I was thinking of aligning this too: 
https://github.com/eclipse/omr/blob/5080dab1ef5fe56e8773ae1655ff98375d7e84ac/include_core/omrport.h#L1410-L1415
and use (only) `OMR_PROCESOR_RV64G` (RV32G is not supported yet anyway)

**Comment by 0xdaryl** [comment]
@genie-omr build aarch64

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**